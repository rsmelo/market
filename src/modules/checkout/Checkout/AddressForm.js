import React from 'react'
import { Field } from 'redux-form'

import InputField from '../../../components/InputField'
import normalizeZipcode from '../../../utils/normalize/normalizeZipcode'


const maskZipcode = '00000-000'

const AddressForm = () => (
  <div>
    <h2>Endereço</h2>
    <label htmlFor="street">Rua</label>
    <Field
      id="street"
      name="street"
      type="input"
      component={InputField}
    />
    <label htmlFor="streetNumber">Número</label>
    <Field
      id="streetNumber"
      name="streetNumber"
      type="input"
      component={InputField}
    />
    <label htmlFor="complementary">Complemento</label>
    <Field
      id="complementary"
      name="complementary"
      type="input"
      component={InputField}
    />
    <label htmlFor="neighborhood">Bairro</label>
    <Field
      id="neighborhood"
      name="neighborhood"
      type="input"
      component={InputField}
    />
    <label htmlFor="city">Cidade</label>
    <Field
      id="city"
      name="city"
      type="input"
      component={InputField}
    />
    <label htmlFor="state">Estado</label>
    <Field
      id="state"
      name="state"
      type="input"
      component={InputField}
    />
    <label htmlFor="zipcode">CEP</label>
    <Field
      id="zipcode"
      name="zipcode"
      type="input"
      normalize={normalizeZipcode}
      placeholder={maskZipcode}
      component={InputField}
    />
  </div>
)

export default AddressForm
