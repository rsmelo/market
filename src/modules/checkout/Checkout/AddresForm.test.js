import React from 'react'
import renderer from 'react-test-renderer'

import AddressForm from './AddressForm'

jest.mock('redux-form', () => ({ Field: 'Field' }))

describe('AddressForm', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<AddressForm />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
