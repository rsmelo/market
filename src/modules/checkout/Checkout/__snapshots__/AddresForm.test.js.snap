// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressForm renders correctly 1`] = `
<div
  className={undefined}
>
  <h2
    className={undefined}
  >
    2. Endereço
  </h2>
  <label
    className={undefined}
    htmlFor="street"
  >
    Rua
  </label>
  <Field
    component={[Function]}
    id="street"
    name="street"
    type="input"
  />
  <label
    className={undefined}
    htmlFor="streetNumber"
  >
    Número
  </label>
  <Field
    component={[Function]}
    id="streetNumber"
    name="streetNumber"
    type="input"
  />
  <label
    className={undefined}
    htmlFor="complementary"
  >
    Complemento
  </label>
  <Field
    component={[Function]}
    id="complementary"
    name="complementary"
    type="input"
  />
  <label
    className={undefined}
    htmlFor="neighborhood"
  >
    Bairro
  </label>
  <Field
    component={[Function]}
    id="neighborhood"
    name="neighborhood"
    type="input"
  />
  <label
    className={undefined}
    htmlFor="city"
  >
    Cidade
  </label>
  <Field
    component={[Function]}
    id="city"
    name="city"
    type="input"
  />
  <label
    className={undefined}
    htmlFor="state"
  >
    Estado
  </label>
  <Field
    component={[Function]}
    id="state"
    name="state"
    type="input"
  />
  <label
    className={undefined}
    htmlFor="zipcode"
  >
    CEP
  </label>
  <Field
    component={[Function]}
    id="zipcode"
    name="zipcode"
    normalize={[Function]}
    placeholder="00000-000"
    type="input"
  />
</div>
`;
